src/webapi/src/controller/user_repository.rs

```diff
@@ -17,14 +22,19 @@ pub async fn insert_user(
     pool: &PgPool,
     user: &CreateUserRequest,
 ) -> Result<(), UserRepositoryError> {
+    let salt = SaltString::generate(&mut OsRng);
+    let hashed_password = Pbkdf2
+        .hash_password(user.password.expose_secret().as_bytes(), &salt)
+        .unwrap()
+        .to_string();
     sqlx::query!(
         r#"
     INSERT INTO users (id, username, password, created_at)
     VALUES ($1, $2, $3, $4)
             "#,
         Uuid::new_v4(),
         user.username,
-        user.password.expose_secret(),
+        hashed_password,
         Utc::now()
     )
     .execute(pool)
```

Comment:
Case in point: how are you planning to verify the password later if you don't store the salt?

Reply:
maybe the naming could be better, but the Pbkdf2 crate returns the password in PHC string format, which contains the salt. See https://docs.rs/pbkdf2/0.12.2/pbkdf2/